sdk/core/Azure.Core/src/ClientOptions.cs

```diff
@@ -90,6 +91,14 @@ public HttpPipelineTransport Transport
         /// </summary>
         public RetryOptions Retry { get; }
 
+        /// <summary>
+        /// The policy to use for retries. If a policy is specified, it will be used in place of the <see cref="Retry"/> property.
+        /// If no policy is specified, a default policy respecting the <see cref="Retry"/> property will be used. The <see cref="RetryPolicy"/> type can be
+        /// derived from to introduce a retry policy that does not require implementing the full retry logic. If <see cref="RetryPolicy.Process"/> is overriden or a
+        /// custom <see cref="HttpPipelinePolicy"/> is specified, it is the implementer's responsibility to populate the <see cref="HttpMessage.ProcessingContext"/> property.
```

Comment:
We're not going to do this for them in a helper policy?

Reply:
Nope - we discussed this and agreed that it is not worth adding a new policy.